rust/src/operations/delete.rs

```diff
@@ -0,0 +1,944 @@
+            }
+
+            if let Some(expr) = &case.else_expr {
+                validate_expr(expr, partition_columns, properties)?;
+            }
+
+            for (when, then) in &case.when_then_expr {
+                validate_expr(when, partition_columns, properties)?;
+                validate_expr(then, partition_columns, properties)?;
+            }
+        }
+        Expr::Cast(cast) => validate_expr(&cast.expr, partition_columns, properties)?,
+        Expr::TryCast(try_cast) => validate_expr(&try_cast.expr, partition_columns, properties)?,
+        Expr::Sort(_) => {
+            return Err(DeltaTableError::Generic(
+                "Sort expression is not allowed Delete operation".to_string(),
+            ))
+        }
+        Expr::ScalarFunction { fun, args } => match fun.volatility() {
+            Volatility::Immutable => {
+                for e in args {
+                    validate_expr(e, partition_columns, properties)?;
+                }
+            }
+            _ => {
+                return Err(DeltaTableError::Generic(
+                    "Nondeterministic functions are not allowed in delete expression".to_string(),
+                ))
+            }
+        },
+        Expr::ScalarUDF { fun, args } => match fun.signature.volatility {
+            Volatility::Immutable => {
+                for e in args {
+                    validate_expr(e, partition_columns, properties)?;
+                }
+            }
+            _ => {
+                return Err(DeltaTableError::Generic(
+                    "Nondeterministic functions are not allowed in delete expression".to_string(),
+                ))
+            }
+        },
+        Expr::AggregateFunction(_) => {
+            return Err(DeltaTableError::Generic(
+                "Aggregate Function is not allowed in delete expression".to_string(),
+            ))
+        }
+        Expr::WindowFunction(_) => {
+            return Err(DeltaTableError::Generic(
+                "Window Function is not allowed in delete expression".to_string(),
+            ))
+        }
+        Expr::AggregateUDF { .. } => {
+            return Err(DeltaTableError::Generic(
+                "Aggregates UDF is not allowed in delete expression".to_string(),
+            ))
+        }
+        Expr::InList { expr, list, .. } => {
+            validate_expr(expr, partition_columns, properties)?;
+            for e in list {
+                validate_expr(e, partition_columns, properties)?;
+            }
+        }
+        Expr::Exists { .. } => {
+            return Err(DeltaTableError::Generic(
+                "Exists expression is not allowed in delete expression".to_string(),
+            ))
+        }
+        Expr::InSubquery { .. } => {
+            return Err(DeltaTableError::Generic(
+                "Subquery expression is not allowed in delete expression".to_string(),
+            ))
+        }
```

Comment:
I think we could probably replace a lot of these
```suggestion
        _ => {
            return Err(DeltaTableError::Generic(format!(
                "Expression is not allowed in delete expression: {}", expr
            )))
        }
```